manager/allocator/networkallocator/networkallocator.go

```diff
@@ -878,3 +890,16 @@ func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	}
 	return s.Spec.Task.Networks
 }
+
+//IsVipOnIngressNetwork check if the vip is in ingress network
+func (na *NetworkAllocator) IsVipOnIngressNetwork(vip *api.Endpoint_VirtualIP) bool {
+	if vip == nil {
+		return false
+	}
+
+	localNet := na.getNetwork(vip.NetworkID)
+	if localNet != nil && localNet.nw != nil {
+		return localNet.nw.Spec.Ingress
```

Comment:
I'm not sure it's necessary to have it defined in `allocator`, as `networkallocator` feels like a more appropriate package, but I'll defer to you. The split between `allocator` and `networkallocator` feels a bit odd to me, since there is a lot of network-specific code in `allocator`. Maybe we should revisit this at some point.